### PR TITLE
Add support for special actions

### PR DESCRIPTION
**Pull request by mattmoor**

### CODE REVIEW

File: k8s/object.bzl [hunk 1]
```diff
@@ -96,9 +96,67 @@ EOF""".format(
...
+./{resolve_script} | \
+  kubectl --cluster="{cluster}" --namespace="{namespace}" replace -f -
+EOF""")
+
+def _describe_impl(ctx):
+  return _common_impl(ctx, """cat > {script} <<"EOF"
+#!/bin/bash -e
+set -o pipefail
+kubectl --cluster="{cluster}" --namespace="{namespace}" describe {kind} \
+  "$(kubectl create --dry-run -f "{unresolved}" | cut -d'"' -f 2)"
```
Comment (dlorenc) [inline k8s/object.bzl]: What's the create --dry-run for?

File: k8s/object.bzl [hunk 2]
```diff
@@ -96,9 +96,67 @@ EOF""".format(
...
+    kwargs["resolve_script"] = ctx.executable.resolved.short_path
+    files += [ctx.executable.resolved]
+    files += list(ctx.attr.resolved.default_runfiles.files)
+
+  if hasattr(ctx.files, "unresolved"):
+    kwargs["unresolved"] = ctx.files.unresolved[0].short_path
+    files += ctx.files.unresolved
+
+  ctx.action(
+      command = template.format(**kwargs),
```
Comment (dlorenc) [inline k8s/object.bzl]: Would it make more sense to use something like: https://docs.bazel.build/versions/master/skylark/lib/actions.html#expand_template

To generate the templated file?

File: k8s/object.bzl (hunk 1 above)
Comment (mattmoor) [reply k8s/object.bzl]: I hoisted this into a variable name.  Essentially I'm abusing this to extract the resource name.  I'm open to suggestions :-/

File: k8s/object.bzl (hunk 2 above)
Comment (mattmoor) [reply k8s/object.bzl]: I moved the string passed from each caller into a file, and now everything goes through `ctx.actions.expand_template` and shares a common implementation :)

**Comment by mattmoor** [comment]
Should be RFAL

**Comment by mattmoor** [comment]
Hmm, I thought I'd disabled concurrent jobs on Travis, but apparently not.  I'm going to close and re-open to confirm it is now properly configured.

**Pull request merged**